src/layouts/AppMenu/AppMenu.client.ts

```diff
@@ -0,0 +1,44 @@
+class AppMenu extends HTMLElement {
+  #menuButton: HTMLButtonElement;
+  #dialog: HTMLDialogElement;
+  #menuList: HTMLElement;
+  #observer: ResizeObserver | undefined;
+
+  constructor() {
+    super();
+    this.#menuButton = this.querySelector('[data-menu-button]') as HTMLButtonElement;
+    this.#dialog = this.querySelector('[data-menu-dialog]') as HTMLDialogElement;
+    this.#menuList = this.querySelector('[data-menu-list]') as HTMLElement;
+  }
+
+  open() {
+    this.#dialog.showModal();
+  }
+
+  close() {
+    this.#dialog.close();
+    this.#menuButton.focus();
+  }
+
+  #onResize() {
+    this.classList.remove('is-compact');
+    const isFitting = this.#menuList.scrollWidth <= this.#menuList.clientWidth;
+    this.classList.toggle('is-compact', !isFitting);
+  }
+
+  connectedCallback() {
+    this.#menuButton.removeAttribute('hidden');
+    this.#menuButton.addEventListener('click', this.open.bind(this));
+    this.#dialog.addEventListener('click', this.close.bind(this));
```

Comment:
I find this behavior a bit weird. Clicking anywhere except the links in the dialog will close the menu dialog. What if I want to copy some text from the menu dialog? That would be impossible at the moment, because the dialog is dismissed the moment I release my mouse/trackpad button. I would expect the close handler to be only on the backdrop and the close button.

Reply:
Will do.

Reply:
Resolved with separate event listener for close button and check if click is outside the dialog.